File: backend/database/Transaction.go
package database

type Transaction struct {
	ID        uint64  `json:"id" gorm:"primary_key"`
	VoucherNo string  `json:"voucherNo" gorm:"not null"`
	Date      string  `json:"date" gorm:"not null"`
	Amount    float64 `json:"amount" gorm:"not null"`
	Contact   string  `json:"contact" gorm:"not null"`
	Cashbook  uint64  `json:"cashbook" gorm:"not null"`
	Reference string  `json:"reference" gorm:"not null"`
	Remarks   string  `json:"remarks" gorm:"not null"`
	Category  string  `json:"category" gorm:"not null"`
}

File: frontend/types/transaction.ts
export type SingleTransactionProperty = {
    id : string
    date : string
    voucherNo : number
    attachments : number
    category : string
    contact : number
    amount: number
    verify : number
    remarks: string
    action : string
}
export type SingleTransactionInput = {
    id: string;
    amount: number | string;
    contact: string;
    remarks: string;
    category: string;
    type: string;
    mode: string;
}
